### ISSUE TITLE
🐛 CSS attr() `<attr-unit>` and `type()` result in errors

### ISSUE THREAD

**Issue by MrSkippy** [biomejs/biome#8030]
### Environment information

```block
> pnpm exec biome rage         
CLI:
  Version:                      2.3.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v22.12.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/9.15.4

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

### What happened?

```css
--percentage: attr(data-percentage %, 100%);
```
```css
--percentage: attr(data-percentage type(<number>), 100);
```

results in an error:

```
  × Unexpected value or character.

...

  i Expected one of:

  - identifier
  - string
  - number
  - dimension
  - ratio
  - custom property
  - function
```

According to the MDN documentation (and working in Chrome) values like `%` or `type(<...>)` should be correct and not throw an error.

[Playground Example](https://biomejs.dev/playground/?cssModules=true&tailwindDirectives=false&code=LwAqACoACgAgACAARQB4AGEAbQBwAGwAZQAgAEgAVABNAEwAOgAKACAAIAA8AGQAaQB2ACAAYwBsAGEAcwBzAD0AIgBsAG8AYQBkAGkAbgBnACAAbABvAGEAZABpAG4AZwAtAC0AdwBpAHQAaAAtAG4AdQBtAGIAZQByACIAIABkAGEAdABhAC0AcABlAHIAYwBlAG4AdABhAGcAZQA9ACIANwA2ACIAPgAKACAAIAAgACAAPABkAGkAdgAgAGMAbABhAHMAcwA9ACIAbABvAGEAZABpAG4AZwBfAF8AYgBhAHIAIgA%2BAGwAbwBhAGQAaQBuAGcAPAAvAGQAaQB2AD4ACgAgACAAPAAvAGQAaQB2AD4ACgAgACAALgAuAC4AIABvAHIAIAB3AGkAdABoACAAYwBsAGEAcwBzACAAYABsAG8AYQBkAGkAbgBnAC0ALQB3AGkAdABoAC0AcABlAHIAYwBlAG4AdABhAGcAZQBgAAoACgAgACAATQBEAE4AIABkAG8AYwB1AG0AZQBuAHQAYQB0AGkAbwBuADoAIABoAHQAdABwAHMAOgAvAC8AZABlAHYAZQBsAG8AcABlAHIALgBtAG8AegBpAGwAbABhAC4AbwByAGcALwBlAG4ALQBVAFMALwBkAG8AYwBzAC8AVwBlAGIALwBDAFMAUwAvAFIAZQBmAGUAcgBlAG4AYwBlAC8AVgBhAGwAdQBlAHMALwBhAHQAdAByACMAcwB5AG4AdABhAHgACgAqAC8ACgAKAC4AbABvAGEAZABpAG4AZwAgAHsACgAgACAALQAtAGIAYQByAC0AYgBnAC0AYwBvAGwAbwByADoAIABsAGkAZwBoAHQAZwByAGEAeQA7AAoAIAAgAC0ALQBiAGEAcgAtAGMAbwBsAG8AcgA6ACAAZwByAGUAZQBuADsACgB9AAoACgAuAGwAbwBhAGQAaQBuAGcALQAtAHcAaQB0AGgALQBuAHUAbQBiAGUAcgAgAHsACgAgACAALQAtAG4AdQBtAGIAZQByADoAIABhAHQAdAByACgAZABhAHQAYQAtAHAAZQByAGMAZQBuAHQAYQBnAGUAIAB0AHkAcABlACgAPABuAHUAbQBiAGUAcgA%2BACkALAAgADEAMAAwACkAOwAKAH0ACgAKAC4AbABvAGEAZABpAG4AZwAtAC0AdwBpAHQAaAAtAHAAZQByAGMAZQBuAHQAYQBnAGUAIAB7AAoAIAAgAC0ALQBwAGUAcgBjAGUAbgB0AGEAZwBlADoAIABhAHQAdAByACgAZABhAHQAYQAtAHAAZQByAGMAZQBuAHQAYQBnAGUAIAAlACwAIAAxADAAMAAlACkAOwAKAH0ACgAKAC4AbABvAGEAZABpAG4AZwAtAC0AdwBpAHQAaAAtAG4AdQBtAGIAZQByACAALgBsAG8AYQBkAGkAbgBnAF8AXwBiAGEAcgAgAHsACgAgACAAYgBhAGMAawBnAHIAbwB1AG4AZAA6ACAAbABpAG4AZQBhAHIALQBnAHIAYQBkAGkAZQBuAHQAKAAKACAAIAAgACAAdABvACAAcgBpAGcAaAB0ACwAIAAKACAAIAAgACAAdgBhAHIAKAAtAC0AYgBhAHIALQBjAG8AbABvAHIAKQAgAGMAYQBsAGMAKAB2AGEAcgAoAC0ALQBuAHUAbQBiAGUAcgApACAAKgAgADEAJQApACwACgAgACAAIAAgAHYAYQByACgALQAtAGIAYQByAC0AYgBnAC0AYwBvAGwAbwByACkAIABjAGEAbABjACgAdgBhAHIAKAAtAC0AbgB1AG0AYgBlAHIAKQAgACoAIAAxACUAKQAKACAAIAApADsACgB9AAoACgAuAGwAbwBhAGQAaQBuAGcALQAtAHcAaQB0AGgALQBwAGUAcgBjAGUAbgB0AGEAZwBlACAALgBsAG8AYQBkAGkAbgBnAF8AXwBiAGEAcgAgAHsACgAgACAAYgBhAGMAawBnAHIAbwB1AG4AZAA6ACAAbABpAG4AZQBhAHIALQBnAHIAYQBkAGkAZQBuAHQAKAAKACAAIAAgACAAdABvACAAcgBpAGcAaAB0ACwAIAAKACAAIAAgACAAdgBhAHIAKAAtAC0AYgBhAHIALQBjAG8AbABvAHIAKQAgAHYAYQByACgALQAtAG4AdQBtAGIAZQByACkALAAKACAAIAAgACAAdgBhAHIAKAAtAC0AYgBhAHIALQBiAGcALQBjAG8AbABvAHIAKQAgAHYAYQByACgALQAtAG4AdQBtAGIAZQByACkACgAgACAAKQA7AAoAfQAKAA%3D%3D&language=css)

### Expected result

Correct CSS `attr()` syntax should not throw an error.

- MDN attr() documentation: https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Values/attr#syntax 
- MDN type() documentation: https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Values/type 

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by Th3S4mur41** [comment]
I'm facing the same problem... Since this is a parser error, there is also no way to work around it with [suppression comment](https://biomejs.dev/analyzer/suppressions/#suppression-syntax)

**Comment by ryan-m-walker** [comment]
I've been doing some recent work on CSS parsing already. If you want some help on this one I could probably take a look.

**Comment by dyc3** [comment]
Went ahead and assigned you @ryan-m-walker, thanks for the interest. Let me know if you want to be unassigned.

**Comment by dyc3** [comment]
duplicate of #6183
